shark_turbine/kernel/ops/wave_ops.py

```diff
@@ -198,6 +206,25 @@ def copy(self, new_name: Optional[str] = None) -> Self:
             new_node.name = new_name
         return get_custom(new_node)
 
+    def copy_to_new_graph(
+        self, new_graph: fx.Graph, new_name: Optional[str] = None
+    ) -> Self:
+        """Returns a duplicate of this node."""
+        new_node = new_graph.node_copy(self.fx_node)
+        new_node.tkw_op = self
+        if new_name:
+            new_node.name = new_name
+        return get_custom(new_node)
+
+    def replace_all_uses_with(self, new_node: CustomOp):
+        """Replace all uses of the current node with the new node."""
+        for user in self.users:
+            user.update_arg(user.node_args.index(self), new_node)
+
+    def erase(self):
+        """Erase the current node from the graph where it exists."""
+        self.graph.erase_node(self.fx_node)
```

Comment:
Eventually we should add additional safety here, e.g. printing a warning if this node still has life uses

Reply:
I can add a warning to handle that case.